gadgets/mri_core/GenericCartesianReconReferencePrepGadget.cpp

```diff
@@ -0,0 +1,301 @@
+            GDEBUG_CONDITION_STREAM(verbose.value(), "Encoding field_of_view : " << e_space.fieldOfView_mm.x << " " << e_space.fieldOfView_mm.y << " " << e_space.fieldOfView_mm.z);
+            GDEBUG_CONDITION_STREAM(verbose.value(), "Recon matrix size : " << r_space.matrixSize.x << " " << r_space.matrixSize.y << " " << r_space.matrixSize.z);
+            GDEBUG_CONDITION_STREAM(verbose.value(), "Recon field_of_view :  " << r_space.fieldOfView_mm.x << " " << r_space.fieldOfView_mm.y << " " << r_space.fieldOfView_mm.z);
+
+            if (!h.encoding[e].parallelImaging)
+            {
+                GDEBUG("Parallel Imaging section not found in header");
+                return GADGET_FAIL;
+            }
+
+            ISMRMRD::ParallelImaging p_imaging = *h.encoding[0].parallelImaging;
+            GDEBUG_CONDITION_STREAM(verbose.value(), "acceFactorE1 is " << p_imaging.accelerationFactor.kspace_encoding_step_1);
+            GDEBUG_CONDITION_STREAM(verbose.value(), "acceFactorE2 is " << p_imaging.accelerationFactor.kspace_encoding_step_2);
+
+            std::string calib = *p_imaging.calibrationMode;
+
+            bool separate = (calib.compare("separate") == 0);
+            bool embedded = (calib.compare("embedded") == 0);
+            bool external = (calib.compare("external") == 0);
+            bool interleaved = (calib.compare("interleaved") == 0);
+            bool other = (calib.compare("other") == 0);
+
+            calib_mode_[e] = Gadgetron::ISMRMRD_noacceleration;
+            if (p_imaging.accelerationFactor.kspace_encoding_step_1 > 1 || p_imaging.accelerationFactor.kspace_encoding_step_2 > 1)
+            {
+                if (interleaved)
+                    calib_mode_[e] = Gadgetron::ISMRMRD_interleaved;
+                else if (embedded)
+                    calib_mode_[e] = Gadgetron::ISMRMRD_embedded;
+                else if (separate)
+                    calib_mode_[e] = Gadgetron::ISMRMRD_separate;
+                else if (external)
+                    calib_mode_[e] = Gadgetron::ISMRMRD_external;
+                else if (other)
+                    calib_mode_[e] = Gadgetron::ISMRMRD_other;
+            }
+        }
+
+        // ---------------------------------------------------------------------------------------------------------
+
+        /*if (!debug_folder.value().empty())
+        {
+            Gadgetron::get_debug_folder_path(debug_folder.value(), debug_folder_full_path_);
+            GDEBUG_CONDITION_STREAM(verbose.value(), "Debug folder is " << debug_folder_full_path_);
+        }
+        else
+        {
+            GDEBUG_CONDITION_STREAM(verbose.value(), "Debug folder is not set ... ");
+        }*/
+
+        return GADGET_OK;
+    }
+
+    int GenericCartesianReconReferencePrepGadget::process(Gadgetron::GadgetContainerMessage< IsmrmrdReconData >* m1)
+    {
+        process_called_times_++;
```

Comment:
Why do we need to know how many times the process has been called? This smells like debug code to me.